src/Geometry/CubeGenerator.h

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file
+ * This file is part of SeisSol.
+ *
+ * @author Sebastian Rettenberger (sebastian.rettenberger AT tum.de,
+ *http://www5.in.tum.de/wiki/index.php/Sebastian_Rettenberger)
+ *
+ * @section LICENSE
+ * Copyright (c) 2015, SeisSol Group
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @section DESCRIPTION
+ **/
+#ifndef CUBEGENERATOR_H
+#define CUBEGENERATOR_H
+
+#include "utils/logger.h"
+#include "utils/args.h"
+#include "MeshReader.h"
+
+#include <omp.h>
+
+#include <algorithm>
+#include <cstring>
+#include <map>
+#include <utility>
+#include <vector>
+
+#include <iostream>
+
+typedef int t_vertex[3];
+
+// changed Vertex to CubeVertex due to same struct naming here and in another part of SeisSol
+struct CubeVertex {
```

Comment:
Is this needed anywhere in the header or in some other places of SeisSol? Otherwise, better to to the cpp file. And enclose with an anonymous namespace then.

Reply:
The `CubeVertex` struct is only used in the `CubeGenerator`. There was a naming clash with other parts of SeisSol which is why I renamed it to `CubeVertex`. 
I'll move it to the source file and enclose it in an anonymous namespace.